tests/test-uart-encoding.toit

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2021 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+import expect show *
+import pixel_strip show *
+import pixel_strip.uart show *
+
+// Test that the Uart code correctly encodes binary data in
+// the high-low pattern that the pixel protocol requires.
+
+class UartTestPixelStrip extends UartEncodingPixelStrip_:
+
+  constructor pixels/int --bytes_per_pixel/int:
+    super pixels --bytes_per_pixel=bytes_per_pixel
+
+  output_interleaved interleaved_data/ByteArray -> none:
+    // We only implemented this method to avoid the abstract class error, it's not used for testing.
+    throw "UNREACHABLE"
+
+expect_uart_equals interleaved/ByteArray array/ByteArray:
+  encoding := create_high_low_encoding interleaved
+  // 3 bits encoded in each byte.
+  bits_encoded := array.size * 3
+
+  // We expect the total number of bits encoded to be a whole byte.
+  expect bits_encoded % 8 == 0
+
+  // We expect each bit to be encoded by three characters (HLL or HHL).
+  expect_equals encoding.size / 3 bits_encoded
+
+  for i := 0; i < bits_encoded; i += 3:
+    byte := array[i / 3]
+    first_bits :=  (byte & 0b00_000_11) >> 0  // Uart transmits low bits first.
```

Comment:
Not sure we should use `_` in other places than 4, 8, ...
I can live with it, though.